ignite/src/main/java/site/ycsb/db/ignite/IgniteClient.java

```diff
@@ -137,63 +125,114 @@ public Status update(String table, String key,
-   * values HashMap will be written into the record with the specified record
-   * key.
-   *
-   * @param table  The name of the table
-   * @param key    The record key of the record to insert.
-   * @param values A HashMap of field/value pairs to insert in the record
-   * @return Zero on success, a non-zero error code on error
-   */
+  /** {@inheritDoc} */
   @Override
-  public Status insert(String table, String key,
-                       Map<String, ByteIterator> values) {
+  public Status insert(String table, String key, Map<String, ByteIterator> values) {
     try {
-      BinaryObjectBuilder bob = cluster.binary().builder("CustomType");
+      if (!table.equals(cacheName)) {
+        throw new UnsupportedOperationException("Unexpected table name: " + table);
+      }
 
-      for (Map.Entry<String, ByteIterator> entry : values.entrySet()) {
-        bob.setField(entry.getKey(), entry.getValue().toString());
+      BinaryObject binObj = convert(values);
 
-        if (debug) {
-          log.info(entry.getKey() + ":" + entry.getValue());
-        }
-      }
+      cache.put(key, binObj);
 
-      BinaryObject bo = bob.build();
+      return Status.OK;
+    } catch (Exception e) {
+      log.error(String.format("Error inserting key: %s", key), e);
+
+      return Status.ERROR;
+    }
+  }
 
-      if (table.equals(DEFAULT_CACHE_NAME)) {
-        cache.put(key, bo);
-      } else {
+  /** {@inheritDoc} */
+  @Override
+  public Status batchInsert(String table, List<String> keys, List<Map<String, ByteIterator>> values) {
+    try {
+      if (!table.equals(cacheName)) {
         throw new UnsupportedOperationException("Unexpected table name: " + table);
       }
 
+      Map<String, BinaryObject> map = new LinkedHashMap<>();
+
+      for (int i = 0; i < keys.size(); i++) {
+        BinaryObject binObj = convert(values.get(i));
+
+        map.put(keys.get(i), binObj);
+      }
+
+      cache.putAll(map);
+
       return Status.OK;
     } catch (Exception e) {
-      log.error(String.format("Error inserting key: %s", key), e);
+      log.error("Error inserting batch of keys.", e);
 
       return Status.ERROR;
     }
   }
 
-  /**
-   * Delete a record from the database.
-   *
-   * @param table The name of the table
-   * @param key   The record key of the record to delete.
-   * @return Zero on success, a non-zero error code on error
-   */
+  /** {@inheritDoc} */
   @Override
   public Status delete(String table, String key) {
     try {
       cache.remove(key);
+
       return Status.OK;
     } catch (Exception e) {
       log.error(String.format("Error deleting key: %s ", key), e);
+
+      return Status.ERROR;
+    }
+  }
+
+  /**
+   * Convert Map<String, ByteIterator> to BinaryObject.
+   *
+   * @param values Values in Map<String, ByteIterator> format.
+   * @return Binary object.
+   */
+  private BinaryObject convert(Map<String, ByteIterator> values) {
+    BinaryObjectBuilder bob = cluster.binary().builder("CustomType");
```

Comment:
Need to check whether building a binary object with the same name N times will create N objects or only 1  object.

Reply:
Oh, please ignore this. I am wrong. After reading the [docs](https://www.gridgain.com/docs/latest/developers-guide/key-value-api/binary-objects) I see that current logic is correct.